doc/reference/actor.md

```diff
@@ -2,755 +2,738 @@
+  identifier will be generated.
+- `ensemble` specifies statically known hosts; it is a hash table mapping server identifiers
+  to lists of addresses.
+  The default known servers only contain the registry with the default registry address.
 
-### protocol-info-id
-::: tip usage
-```
-(protocol-info-id ...)
+### stop-actor-server!
+```scheme
+(stop-actor-server! (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Stops and joins an actor server.
 
-### protocol-info-extend
-::: tip usage
+### actor-server-identifier
+```scheme
+(actor-server-identifier (srv (current-actor-server)))
 ```
-(protocol-info-extend ...)
-```
-:::
 
-Please document me!
+Returns an actor server's identifier.
 
-### protocol-info-calls
-::: tip usage
-```
-(protocol-info-calls ...)
+### register-actor!
+```scheme
+(register-actor! name (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Registers the current thread in an actor server as an actor with the name `name`.
 
-### protocol-info-events
-::: tip usage
-```
-(protocol-info-events ...)
+### connect-to-server!
+```scheme
+(connect-to-server! id (addrs #f) (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Instructs an actor server to connect to another server.
+- `id` is the identifier of the target server
+- `addrs` is an optional list of addresses; if none is specified and
+  the server is uknown, its addresses will be resolved through the
+  ensemble registry.
 
-### protocol-info-streams
-::: tip usage
+### list-actors
+```scheme
+(list-actors (srv (current-actor-server)))
 ```
-(protocol-info-streams ...)
-```
-:::
-
-Please document me!
 
+Lists the actors registered with an actor server.
+Returns a list of references.
 
-## RPC
-
-### current-rpc-server
-::: tip usage
-```
-(current-rpc-server ...)
+### list-connections
+```scheme
+(list-connections (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Lists the current connections of an actor server.
+Returns an associative list, with the server identifier at the car and the
+list of addresses connected at the cdr.
 
-### start-rpc-server!
-::: tip usage
+### default-known-servers
+```scheme
+(default-known-servers)
 ```
-(start-rpc-server! ...)
-```
-:::
 
-Please document me!
+Returns a hash table with the default known servers.
+By default it only contains the registry reachable in its default address.
 
-### stop-rpc-server!
-::: tip usage
-```
-(stop-rpc-server! ...)
+### set-default-known-servers!
+```scheme
+(set-default-known-servers! servers)
 ```
-:::
 
-Please document me!
+Sets the default known servers; `servers` must be a hash table mapping
+server identifiers to lists of addresses.
 
-### rpc-connect
-::: tip usage
-```
-(rpc-connect ...)
+### default-registry-addresses
+```scheme
+(default-registry-addresses)
 ```
-:::
 
-Please document me!
+Returns the default registry addresses, as a list of addresses.  By
+default the registry is reachable at `/tmp/ensemble/registry` in the
+current host.
```

Comment:
Yes, it is an actual unix domain socket.

No, per user sockets are a mistake, ensembles might be using different users for different servers for privilege separation at the unix level.
If we make it per user, we lose that capability.

Reply:
I don't think I need to say anything else here, I don't understand how you find this confusing.

Reply:
When using different servers with privilege separation, it is easy to explicitly point the socket at a shared location. But since the default is current user, the default should be under a path controlled by current user.

Reply:
I disagree here.

System time sharing by multiple (human) users is a thing of the 20th century. It can still happen, but it is exceedingly rare.

What we have in the 21st century is user workstations and laptops, and servers in the cloud where the user abstraction is just a privilege separation mechanism.
So I strongly think that the current default is the right thing to do.